daemon/cmd/local_node_sync.go

```diff
@@ -120,16 +124,45 @@ func (ini *localNodeSynchronizer) getK8sLocalNode(ctx context.Context) (*slim_co
 	return nil, ctx.Err()
 }
 
+func (ini *localNodeSynchronizer) getK8sLocalCiliumNode(ctx context.Context) (*v2.CiliumNode, error) {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	for ev := range ini.K8sCiliumLocalNode.Events(ctx) {
+		ev.Done(nil)
+		if ev.Kind == resource.Upsert {
+			return ev.Object, nil
+		}
+	}
+	return nil, ctx.Err()
+}
+
 func (ini *localNodeSynchronizer) initFromK8s(ctx context.Context, node *node.LocalNode) error {
-	if ini.K8sLocalNode == nil {
+	if ini.K8sLocalNode == nil || ini.K8sCiliumLocalNode == nil {
 		return nil
 	}
 
 	k8sNode, err := ini.getK8sLocalNode(ctx)
 	if err != nil {
 		return err
 	}
-	parsedNode := parseNode(k8sNode)
+	parsedNode := k8s.ParseNode(k8sNode, source.Kubernetes)
+
+	ctx, cancel := context.WithTimeout(ctx, time.Second)
+	defer cancel()
+	// In cases where no local CiliumNode exists (such as on a fresh node) we skip restoring
+	// the CiliumNode information from k8s.
+	k8sCiliumNode, err := ini.getK8sLocalCiliumNode(ctx)
+	if err != nil {
+		if errors.Is(err, context.DeadlineExceeded) {
+			log.Info("No CiliumNode found in k8s, skipping restoring CiliumNode information from k8s, assuming fresh node")
+			return nil
+		}
+		return fmt.Errorf("failed to get local ciliumnode from store: %w", err)
+	}
+
+	if parsedNode.Name != k8sCiliumNode.Name {
+		return fmt.Errorf("ciliumnode name %q does not match node name %q", k8sCiliumNode.Name, parsedNode.Name)
+	}
```

Comment:
This check is likely superfluous, given that the LocalNodeResource already enforces that the name matches: https://github.com/tommyp1ckles/cilium/blob/pr%2Ftp%2Frestore-cilium-host-ip-from-cn/daemon/k8s/resources.go#L53-L68

Reply:
yeah makes sense, ill remove this.